lib/metanorma/collection_manifest/collection_manifest.rb

```diff
@@ -0,0 +1,231 @@
+      end
+    end
+
+    def manifest_filexist(config, dir)
+      if config.file
+        file = @collection.class.resolve_fileref(dir, config.file)
+        @collection.class.check_file_existence(file)
+        config.file = Pathname.new(file).relative_path_from(Pathname.new(dir))
+      end
+      Array(config.entry).each do |f|
+        manifest_filexist(f, dir)
+      end
+    end
+
+    def manifest_expand_yaml(config, dir)
+      Array(config.entry).each do |e|
+        currdir = dir
+        /\.ya?ml$/.match?(e.file) and
+          currdir = manifest_expand_yaml_entry(e, dir)
+        manifest_expand_yaml(e, currdir)
+      end
+      config
+    end
+
+    def manifest_expand_yaml_entry(entry, dir)
+      f = @collection.class.resolve_fileref(dir, entry.file)
+      currdir = File.dirname(f)
+      @collection.class.check_file_existence(f)
+      entry.file = nil
+      entry.entry = CollectionConfig::Config.from_yaml(File.read(f)).manifest
+      if currdir != dir
+        prefix = Pathname.new(currdir).relative_path_from(Pathname.new(dir))
+        update_filepaths(entry.entry, prefix.to_s)
+      end
+      currdir
+    end
+
+    def update_filepaths(entry, prefix)
+      entry.file && !(Pathname.new entry.file).absolute? and
+        entry.file = File.join(prefix, entry.file)
+      entry.entry.each do |f|
+        update_filepaths(f, prefix)
+      end
+    end
+
+    def manifest_compile_adoc(config, dir)
+      if /\.adoc$/.match?(config.file)
+        file = @collection.class.resolve_fileref(dir, config.file)
+        config.file = compile_adoc(dir, file, config.file)
+      end
+      Array(config.entry).each do |f|
+        manifest_compile_adoc(f, dir)
+      end
+    end
+
+    def compile_adoc(dir, resolved_filename, rel_filename)
+      compile_adoc_file(dir, resolved_filename)
+      set_adoc2xml(rel_filename)
+    end
+
+    # @param fileref [String]
+    def set_adoc2xml(fileref)
+      File.join(
+        File.dirname(fileref),
+        File.basename(fileref).gsub(/.adoc$/, ".xml"),
+      )
+    end
+
+    # param filepath [String]
+    # @raise [AdocFileNotFoundException]
+    def compile_adoc_file(dir, file)
+      f = (Pathname.new file).absolute? ? file : File.join(dir, file)
+      unless File.exist? f
+        raise AdocFileNotFoundException.new "#{f} not found!"
+      end
+
+      Util.log("[metanorma] Info: Compiling #{f}...", :info)
+      Metanorma::Compile.new
+        .compile(f, agree_to_terms: true, no_install_fonts: true,
+                    extension_keys: [:xml])
+      Util.log("[metanorma] Info: Compiling #{f}...done!", :info)
+    end
+
+    def documents(dir = "", mnf = @config)
+      Array(mnf.entry).each_with_object({}) do |dr, m|
+        if dr.file
+          m[Util::key dr.identifier] = documents_add(dir, dr)
+        elsif dr.entry
+          m.merge! documents(dir, dr)
+        end
+        m
+      end
+    end
+
+    def documents_add(dir, docref)
```

Comment:
There should not be a need to pass `dir` because this object should track where itself is, via `@location` or some variable.

Reply:
Done